### PR TITLE
don't segfault on unreadable dir

### PR DESCRIPTION
**Pull request by ptarjan**
I ran into this in some of my testing

### Test plan

See included automated tests.

### CODE REVIEW

File: common/common.cc [hunk 1]
```diff
@@ -252,8 +252,11 @@ void appendFilesInDir(string_view basePath, string_view path, const sorbet::Unor
             if (!recursive || strcmp(entry->d_name, ".") == 0 || strcmp(entry->d_name, "..") == 0) {
                 continue;
             }
-            appendFilesInDir(basePath, fullPath, extensions, recursive, result, absoluteIgnorePatterns,
-                             relativeIgnorePatterns);
+            try {
+                appendFilesInDir(basePath, fullPath, extensions, recursive, result, absoluteIgnorePatterns,
+                                 relativeIgnorePatterns);
+            } catch (sorbet::FileNotFoundException) {
+            }
```
Comment (DarkDimius) [inline common/common.cc]: can we report an error here rather than swallow the exception?

**Comment by ptarjan** [comment]
re-r? @dmitry-stripe

**Comment by ptarjan** [comment]
wtf... does linux not fail with `readdir` on a `chmod 000` dir?  or are we running as root?

**Comment by DarkDimius** [comment]
> wtf... does linux not fail with readdir on a chmod 000 dir? or are we running as root?

afaik inside docker container we're root

**Pull request merged**